contracts/batcher/TwoWayBatcher.sol

```diff
@@ -0,0 +1,85 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.13;
+
+import "hardhat/console.sol";
+import "@equilibria/root/number/types/UFixed18.sol";
+import "@equilibria/root/token/types/Token18.sol";
+import "@equilibria/root/token/types/Token6.sol";
+import "@equilibria/root/control/unstructured/UReentrancyGuard.sol";
+import "./Batcher.sol";
+
+contract TwoWayBatcher is UReentrancyGuard, Batcher {
+    using UFixed18Lib for UFixed18;
+    using Token18Lib for Token18;
+    using Token6Lib for Token6;
+
+    event USDCLoaned(address indexed depositor, UFixed18 amount);
+    event USDCRepaid(address indexed depositor, UFixed18 amount);
+
+    error TwoWayBatcherInvalidUSDCAmount(UFixed18 amount);
+
+    UFixed18 public usdcLoansOutstanding;
+    mapping(address => UFixed18) public depositorToUSDCLoanAmount;
+
+    constructor(IEmptySetReserve reserve, Token18 dsu, Token6 usdc)
+    Batcher(reserve, dsu, usdc)
+    { }
+
+    function loanUSDC(UFixed18 amount) external nonReentrant {
+        rebalance();
+
+        if (!amount.eq(_toToken6Amount(amount))) revert TwoWayBatcherInvalidUSDCAmount(amount);
+        USDC.pull(msg.sender, amount, true);
+
+        usdcLoansOutstanding = usdcLoansOutstanding.add(amount);
+        depositorToUSDCLoanAmount[msg.sender] = depositorToUSDCLoanAmount[msg.sender].add(amount);
+
+        emit USDCLoaned(msg.sender, amount);
+    }
+
+    function repayUSDC(UFixed18 amount) external nonReentrant {
+        rebalance();
+
+        if (!amount.eq(_toToken6Amount(amount))) revert TwoWayBatcherInvalidUSDCAmount(amount);
+        usdcLoansOutstanding = usdcLoansOutstanding.sub(amount);
+        depositorToUSDCLoanAmount[msg.sender] = depositorToUSDCLoanAmount[msg.sender].sub(amount);
+
+        USDC.push(msg.sender, amount);
+
+        emit USDCRepaid(msg.sender, amount);
+    }
+
+    function _rebalance(UFixed18 usdcBalance, UFixed18) override internal {
+        uint256 cmpResult = usdcLoansOutstanding.compare(usdcBalance);
```

Comment:
very nit: I think it's slightly easier to follow the logic if this is flipped:

`uint256 balanceToTarget = ....`, 0 -> balance too low, redeem, redeem, 1 -> good, 2 -> balance too high, supply